api/lib/infrastructure/repositories/knowledge-element-repository.js

```diff
@@ -36,57 +19,67 @@ function _findByCampaignIdForSharedCampaignParticipationWhere(campaignParticipat
     .where({ 'campaign-participations.isShared': true })
     .where(campaignParticipationsWhereClause)
     .where({ status: 'validated' })
-    .fetchAll()
-    .then(_toDomain);
+    .fetchAll();
+
+  return bookshelfToDomainConverter.buildDomainObjects(BookshelfKnowledgeElement, keResults);
 }
 
 module.exports = {
 
-  save(knowledgeElement) {
-
-    return Promise.resolve(_.omit(knowledgeElement, ['id', 'createdAt']))
-      .then((knowledgeElement) => new BookshelfKnowledgeElement(knowledgeElement))
-      .then((knowledgeElementBookshelf) => knowledgeElementBookshelf.save())
-      .then(_toDomain);
-  },
-
-  findUniqByUserId({ userId, limitDate }) {
-    return BookshelfKnowledgeElement
-      .query((qb) => {
-        qb.where({ userId });
+  _getMostRecentPerSkillExcludingResettedByUserIdAndLimitDateQuery({ userId, limitDate }) {
```

Comment:
```suggestion
  _getMostRecentPerSkillExcludingResetByUserIdAndLimitDateQuery({ userId, limitDate }) {
```
`resetted` n'est pas anglais 😛

Reply:
sure my dear ;)